File: app/src/main/java/com/innerly/app/ui/GoalAdapter.java
package com.innerly.app.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.innerly.app.R;
import com.innerly.app.data.Goal;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class GoalAdapter extends RecyclerView.Adapter<GoalAdapter.GoalViewHolder> {

    // Click එකක් හඳුනා ගැනීමට Interface එකක් සාදා ගැනීම
    public interface OnGoalClickListener {
        void onGoalClick(Goal goal);
    }

    private List<Goal> goals = new ArrayList<>();
    private final OnGoalClickListener listener;

    // Constructor
    public GoalAdapter(OnGoalClickListener listener) {
        this.listener = listener;
    }

    public void setGoals(List<Goal> newGoals) {
        this.goals = newGoals;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public GoalViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_goal, parent, false);
        return new GoalViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull GoalViewHolder holder, int position) {
        Goal goal = goals.get(position);
        holder.bind(goal);

        // අයිතමය ක්ලික් කළ විට listener එක ක්‍රියාත්මක කිරීම
        holder.itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.onGoalClick(goal);
            }
        });
    }

    @Override
    public int getItemCount() {
        return goals.size();
    }

    // ViewHolder Class එක
    public static class GoalViewHolder extends RecyclerView.ViewHolder {
        private final TextView tvTitle;
        private final TextView tvDescription;
        private final TextView tvMood;
        private final TextView tvDate;

        public GoalViewHolder(@NonNull View itemView) {
            super(itemView);
            tvTitle = itemView.findViewById(R.id.tvGoalTitle);
            tvDescription = itemView.findViewById(R.id.tvGoalDescription);
            tvMood = itemView.findViewById(R.id.tvMoodLabel);
            tvDate = itemView.findViewById(R.id.tvUpdatedDate);
        }

        public void bind(Goal goal) {
            tvTitle.setText(goal.getTitle());
            tvDescription.setText(goal.getDescription());
            tvMood.setText(goal.getCurrentMood());

            SimpleDateFormat sdf = new SimpleDateFormat("MMM dd, yyyy", Locale.getDefault());
            String formattedDate = sdf.format(new Date(goal.getCreatedDate()));

            // String සම්පත් (Resources) ලබා ගැනීම
            String dateText = itemView.getContext().getString(R.string.goal_created_date, formattedDate);
            tvDate.setText(dateText);
        }
    }
}

File: app/src/main/java/com/innerly/app/ui/GoalAdapter.kt
package com.innerly.app.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.innerly.app.R
import com.innerly.app.data.Goal
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class GoalAdapter(private val onGoalClick: (Goal) -> Unit) : RecyclerView.Adapter<GoalAdapter.GoalViewHolder>() {

    private var goals: List<Goal> = emptyList()

    fun setGoals(newGoals: List<Goal>) {
        this.goals = newGoals
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GoalViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_goal, parent, false)
        return GoalViewHolder(view)
    }

    override fun onBindViewHolder(holder: GoalViewHolder, position: Int) {
        val goal = goals[position]
        holder.bind(goal)
        holder.itemView.setOnClickListener { onGoalClick(goal) }
    }

    override fun getItemCount(): Int = goals.size

    class GoalViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvTitle: TextView = itemView.findViewById(R.id.tvGoalTitle)
        private val tvDescription: TextView = itemView.findViewById(R.id.tvGoalDescription)
        private val tvMood: TextView = itemView.findViewById(R.id.tvMoodLabel)
        private val tvDate: TextView = itemView.findViewById(R.id.tvUpdatedDate)

        fun bind(goal: Goal) {
            tvTitle.text = goal.title
            tvDescription.text = goal.description
            tvMood.text = goal.currentMood
            
            val sdf = SimpleDateFormat("MMM dd, yyyy", Locale.getDefault())
            val formattedDate = sdf.format(Date(goal.createdDate))
            tvDate.text = itemView.context.getString(R.string.goal_created_date, formattedDate)
        }
    }
}